mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -556,6 +571,7 @@ class Word2VecModel private[spark] (
       .sortBy(- _._2)
       .take(num + 1)
       .tail
+      .map(v => (if (vecNorm == 0) v else (v._1, v._2 / vecNorm)))
```

Comment:
Strictly speaking we could throw an error or warning if passing in a zero vector as it's meaningless as you say. Still, I think in real usage that's unlikely, so I'd just go for the first option. I don't think there are real world efficiency gains to be had with the second option.